src-tauri/src/lib.rs

```diff
@@ -50,14 +51,49 @@ fn is_debug() -> bool {
 #[cfg_attr(mobile, tauri::mobile_entry_point)]
 pub fn run() {
     tauri::Builder::default()
+        .plugin(tauri_plugin_single_instance::init(|app, _args, _cwd| {
+            let _ = app
+                .get_webview_window("main")
+                .expect("no main window")
+                .set_focus();
+        }))
+        .plugin(tauri_plugin_deep_link::init())
         .plugin(tauri_plugin_http::init())
         .plugin(tauri_plugin_dialog::init())
         .manage(Mutex::new(None) as DebuggerBody)
         .manage(Arc::new(Mutex::new(FlushDisplayState::default())) as FlushDisplayBody)
         .manage(Mutex::new(MidiProviderContainer::None))
         .menu(create_menu)
         .setup(|app| {
-            app.manage(AccessManager::load(app.handle().clone()));
+            let map_paths = |paths: Vec<Url>| {
+                paths
+                    .into_iter()
+                    .flat_map(|url| {
+                        if url.scheme() == "file" {
+                            url.to_file_path().ok()
+                        } else {
+                            None
+                        }
+                    })
+                    .collect::<Vec<_>>()
+            };
+
+            let access_manager = AccessManager::load(app.handle().clone());
+
+            let deep_link = app.deep_link();
```

Comment:
Last time I saw the tauri file association example, they had a section where they read from command line arguments to handle Windows links. I wonder if this will work on Windows.

But if it works on macOS that's a step up anyway. I think it's worth merging, and if there's any hiccups, patch it after.